typing/typedecl.ml

```diff
@@ -1911,24 +1911,24 @@ let abstract_type_decl ~injective arity =
     }
   end
 
-let approx_type_decl sdecl_list =
+let approx_type_decl sdecl_list ~abstract_origin =
```

Comment:
Nitpicking: as `abstract_origin` is an auxillary parameter, I would propose to move it before the main argument `sdecl_list`.